backend/RPI_Smart_Calendar/Calendar/views.py

```diff
@@ -190,9 +211,22 @@ def post(self, request, *args, **kwargs):
             output["isSuccess"] = False
             output["Message"] = 'FAIL: PLEASE LOGIN'
             return JsonResponse(status=401, data = output, safe=False)
+        # get group
+        group = None
+        if not isblock:
+            try:
+                group = MyGroup.objects.all().get(group_id = groupid)
+            except:
+                output["isSuccess"] = False
+                output["Message"] = 'FAIL: USER IS NOT IN THE GROUP'
+                return JsonResponse(status=401, data = output, safe=False)
         #save event
-        event = Event(user = user, title = title, startTime = startTime, endTime = endTime,
-                        method = 'manually', type = 'block', details = details)
+        if isblock:
```

Comment:
Actually, I feel that a time point can also have `details`. So, we do not need this `if ... else ... `